packages/ra-core/src/RoutesWithLayout.js

```diff
@@ -16,17 +16,7 @@ const RoutesWithLayout = ({
 
     return (
         <Switch>
-            {customRoutes &&
-                customRoutes.map((route, index) => (
-                    <Route
-                        key={index}
-                        exact={route.props.exact}
-                        path={route.props.path}
-                        component={route.props.component}
-                        render={route.props.render}
-                        children={route.props.children} // eslint-disable-line react/no-children-prop
-                    />
-                ))}
+            {customRoutes && customRoutes.map((route) => cloneElement(route))}
```

Comment:
Are you sure that React won't complain about missing `key` here? Did you test that code?

Reply:
Yes I did, it does not through any warning.

Reply:
Well, I should have checked before merging: your code does throw a warning in the simple demo:

```
warning.js:33 Warning: Each child in an array or iterator should have a unique "key" prop.

Check the render method of `CoreAdminRouter`. See https://fb.me/react-warning-keys for more information.
    in Route
    in CoreAdminRouter (created by Connect(CoreAdminRouter))
    in Connect(CoreAdminRouter) (created by getContext(Connect(CoreAdminRouter)))
    in getContext(Connect(CoreAdminRouter)) (created by Route)
    in Route (created by CoreAdmin)
    in Switch (created by CoreAdmin)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by CoreAdmin)
    in TranslationProviderView (created by Connect(TranslationProviderView))
    in Connect(TranslationProviderView) (created by CoreAdmin)
    in Provider (created by CoreAdmin)
    in CoreAdmin (created by withContext(CoreAdmin))
    in withContext(CoreAdmin)
```

I'll have to fix that in next.